src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -49,6 +49,43 @@ def iterated_filtsmooth(
         stopcrit: StoppingCriterion
             A stopping criterion for iterated filtering.
 
+        Returns
+        -------
+        SmoothingPosterior
+
+        See Also
+        --------
+        RegressionProblem: a regression problem data class
+        """
+
+        result = None
+        for iter_result in self.iterated_filtsmooth_generator(
+            regression_problem, stopcrit
+        ):
+            result = iter_result
+        return result
```

Comment:
If you have a more elegant solution, feel free to request changes.
I thought about something like

```python
return next(reversed(self.iterated_filtsmooth_generator(...)))
```

but I think it might be less efficient. If efficiency is relevant at all here ....

Reply:
I am not sure about your second proposal. I like the for loop, but you dont need the `result = iter_result` assignment I think. ```for iter_result in self.iterated_filtsmooth_generator(...): pass; return iter_result``` would work, right? 

Apparently (https://stackoverflow.com/questions/2138873/cleanest-way-to-get-last-item-from-python-iterator) ```*_, last = self.iterated_filtsmooth_generator(...)``` would also be an option? I have not checked this though

Reply:
checked it, the `*_, last = ...` thing works :)

Reply:
Okay. I like the second proposal! The first, with the `for ... pass ...` loop doesn't work -- that I checked.

Reply:
Okay nevermind, your first proposal works, pylint was just telling me that it isn't a good idea. However, I would ignore pylint in this case and use the first proposal because this way, the objects aren't kept in memory.